File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/example/adapter/ExampleAdapter.java
package com.example.stas.oxforddictionary.presentation.view.example.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.example.stas.oxforddictionary.R;
import java.util.List;
import butterknife.BindView;
import butterknife.ButterKnife;

public class ExampleAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private final List<String> examples;
    private final Context context;

    public ExampleAdapter(List<String> examples, Context context) {
        this.examples = examples;
        this.context = context;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.recycler_examples_example_item, viewGroup, false);
        return new ExampleViewHolder(view);
    }
    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int i) {
        ((ExampleViewHolder)viewHolder).bind(examples.get(i));
    }
    @Override
    public int getItemCount() {
        return examples.size();
    }

    class ExampleViewHolder extends RecyclerView.ViewHolder{
            @BindView(R.id.exampleTW) TextView exampleTW;
        ExampleViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
        void bind(String example){
            exampleTW.setText(context.getString(R.string.example_quotes, example));
        }
    }
}

File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/example/adapter/ExampleAdapter.kt
package com.example.stas.oxforddictionary.presentation.view.example.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.example.stas.oxforddictionary.R

class ExampleAdapter(private val examples: List<String?>, private val context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.recycler_examples_example_item, viewGroup, false)
        return ExampleViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, i: Int) {
        (viewHolder as ExampleViewHolder).bind(examples[i])
    }

    override fun getItemCount(): Int {
        return examples.size
    }

    internal inner class ExampleViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @BindView(R.id.exampleTW)
        lateinit var exampleTW: TextView

        init {
            ButterKnife.bind(this, itemView)
        }

        fun bind(example: String?) {
            exampleTW.text = context.getString(R.string.example_quotes, example)
        }
    }
}
